### PR TITLE
Implement shader-interpreter relative `MOVA` addressing

### PR DESCRIPTION
**Pull request by Wunkolo**
pando

### CODE REVIEW

File: src/core/PICA/shader_interpreter.cpp [hunk 1]
```diff
@@ -303,7 +312,6 @@ void PICAShader::mova(u32 instruction) {
 	const u32 src = getBits<12, 7>(instruction);
 	const u32 idx = getBits<19, 2>(instruction);
 
-	if (idx) Helpers::panic("[PICA] MOVA: idx != 0");
```
Comment (wheremyfoodat) [inline src/core/PICA/shader_interpreter.cpp:306]: Shouldn't this be using getIndexedSource before getSourceSwizzled?

Comment (Wunkolo) [reply src/core/PICA/shader_interpreter.cpp:306]: Oop you're right. I though loading a value into the address register didn't use the previous address register.

File: src/core/PICA/shader_interpreter.cpp [hunk 2]
```diff
@@ -130,15 +138,16 @@ u8 PICAShader::getIndexedSource(u32 source, u32 index) {
...
-	else if (source < 0x20)
+	} else if (source < 0x20) {
 		return tempRegisters[source - 0x10];
-	else if (source <= 0x7f)
-		return floatUniforms[source - 0x20];
-	else {
-		Helpers::warn("[PICA] Unimplemented source value: %X\n", source);
-		return vec4f({f24::zero(), f24::zero(), f24::zero(), f24::zero()});
+	} else {
+		const usize floatIndex = (source - 0x20) & 0x7f;
```
Comment (wheremyfoodat) [inline src/core/PICA/shader_interpreter.cpp:146]: Is the wrapping behavior that the & 0x7f incurs confirmed somewhere 🤔

Comment (Wunkolo) [reply src/core/PICA/shader_interpreter.cpp:146]: Confirmed here: https://www.3dbrew.org/wiki/GPU/Shader_Instruction_Set#Relative_addressing

**Pull request merged**